state-chain/runtime/src/chainflip/vault_swap.rs

```diff
@@ -109,3 +119,134 @@ pub fn bitcoin_vault_swap(
 		deposit_address,
 	})
 }
+
+pub fn solana_vault_swap(
+	broker_id: AccountId,
+	source_asset: Asset,
+	destination_asset: Asset,
+	destination_address: EncodedAddress,
+	broker_commission: BasisPoints,
+	extra_parameters: VaultSwapExtraParametersEncoded,
+	channel_metadata: Option<CcmChannelMetadata>,
+	boost_fee: BasisPoints,
+	affiliate_fees: Affiliates<AccountId>,
+	dca_parameters: Option<DcaParameters>,
+) -> Result<VaultSwapDetails<String>, DispatchErrorWithMessage> {
+	// Load up environment variables.
+	let api_environment = SolEnvironment::api_environment().map_err(|_| {
+		DispatchErrorWithMessage::Other("Failed to load Solana API environment".into())
```

Comment:
Avoid using `DispatchErrorWithMessage::Other("Error string".into())`: The string gets converted to `DispatchError::Other`, which erases the string... 

(See #5485, maybe copy the changes from there)

Reply:
I've cherry-picked the change into this feature branch and made the appropriate changes.